src/commands/generateThought.ts

```diff
@@ -11,9 +11,56 @@ import getThoughtById from '../selectors/getThoughtById'
 import simplifyPath from '../selectors/simplifyPath'
 import head from '../util/head'
 import isDocumentEditable from '../util/isDocumentEditable'
+import isURL from '../util/isURL'
 import parentOf from '../util/parentOf'
 import pathToContext from '../util/pathToContext'
 
+/** Fetches the title of a webpage from its URL. */
+const fetchWebpageTitle = async (url: string): Promise<string | null> => {
+  try {
+    // Ensure the URL has a protocol
+    const fullUrl = url.startsWith('http') ? url : `https://${url}`
+
+    const response = await fetch(fullUrl, {
+      method: 'GET',
+      mode: 'cors',
+      headers: {
+        Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+      },
+    })
+
+    if (!response.ok) {
+      return null
+    }
+
+    const html = await response.text()
+
+    // Extract title from HTML
+    const titleMatch = html.match(/<title[^>]*>([^<]*)<\/title>/i)
+    if (titleMatch && titleMatch[1]) {
+      // Decode HTML entities in the title content
+      const rawTitle = titleMatch[1].trim()
+      // Decode HTML entities manually for the most common ones
+      const decodedTitle = rawTitle
+        .replace(/&amp;/g, '&')
+        .replace(/&lt;/g, '<')
+        .replace(/&gt;/g, '>')
+        .replace(/&quot;/g, '"')
+        .replace(/&#39;/g, "'")
+        .replace(/&nbsp;/g, ' ')
+
+      // Replace < and > with ( and ) to avoid HTML tag conflicts in the thought system
+      const cleanTitle = decodedTitle.replace(/</g, '(').replace(/>/g, ')')
+
+      return cleanTitle
+    }
+
+    return null
+  } catch (err) {
```

Comment:
Never swallow error messages. Allow it to bubble up and catch it above where an error action can be dispatched.